### PR TITLE
Fix eyaml secret configuration

### PR DESCRIPTION
**Pull request by davidphay**
- feat: deploy only necessary configuration regarding the solution (secret, keys stored in values.yaml, configmap )
- feat deploy only the most secure configuration (secret > keys stored in values.yaml > configMap)
- feat: generate a warning if configmap or keys stored in values.yaml is use
- feat: generate an error if keys stored in values.yaml is use and if `.Values.eyaml.public_key` or `.Values.eyaml.private_key` is missing

### CODE REVIEW

File: templates/puppetserver-statefulset-compilers.yaml [hunk 1]
```diff
@@ -67,16 +67,14 @@ spec:
               chmod +x /etc/puppetlabs/puppet/r10k_hiera_entrypoint.sh /etc/puppetlabs/puppet/r10k_hiera_cronjob.sh;
               {{- end }}
               {{- if .Values.hiera.config }}
-              cp /etc/puppetlabs/puppet/configmap/hiera.yaml /etc/puppetlabs/puppet/hiera.yaml;
+              cp /etc/puppetlabs/puppet/configmap/hiera.ya(not .Values.hiera.eyaml.existingSecret)ml /etc/puppetlabs/puppet/hiera.yaml;
```
Comment (Xtigyro) [inline templates/puppetserver-statefulset-compilers.yaml]: Please fix the if statement here. It's on the wrong line.

**Review by Xtigyro — CHANGES_REQUESTED** [review]
Thanks for the contribution once again, @davidphayanouvong! 🥇

Could you please update the `CHANGELOG.md` following the same pattern for previous releases and bump this time around the `minor` chart version in `Chart.yaml`?

**Comment by Xtigyro** [comment]
Could you please update the `CHANGELOG.md` following the same pattern for previous releases and bump the `minor` chart version in `Chart.yaml`?

File: templates/puppetserver-statefulset-compilers.yaml (hunk 1 above)
Comment (davidphay) [reply templates/puppetserver-statefulset-compilers.yaml]: Fixed

**Review by Xtigyro — APPROVED** [review]
Thanks once more, @davidphayanouvong!

**Comment by Xtigyro** [comment]
@davidphayanouvong Could you please sign the CLA?

**Comment by davidphay** [comment]
> @davidphayanouvong Could you please sign the CLA?

Its ok for me, but I think you have to sign rhw CLA too ?

**Comment by Xtigyro** [comment]
@davidphayanouvong Ha-ha-ha-ha, that was fun! You were so right!

**Pull request merged**